extensions/healthie/actions/sendFormCompletionRequest.ts

```diff
@@ -26,6 +26,72 @@ const fields = {
     type: FieldType.STRING,
     required: true,
   },
+  is_recurring: {
+    id: 'is_recurring',
+    label: 'Is recurring',
+    description: 'Set to true if the Form completion should be recurring.',
+    type: FieldType.BOOLEAN,
+    required: false,
+  },
+  frequency: {
+    id: 'frequency',
+    label: 'Frequency',
+    description:
+      'Required if "Is recurring" is set to true. Valid options are: Daily, Weekly, Monthly.',
+    type: FieldType.STRING,
+    required: false,
+  },
+  period: {
+    id: 'period',
+    label: 'Period',
+    description: 'AM or PM.',
+    type: FieldType.STRING,
+    required: false,
+  },
+  hour: {
+    id: 'hour',
+    label: 'Hour',
+    description:
+      'For instance, if you want to trigger the completion request at 1:05 PM, use "1".',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  minute: {
+    id: 'minute',
+    label: 'Minute',
+    description:
+      'For instance, if you want to trigger the completion request at 1:05 PM, use "5".',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  weekday: {
+    id: 'weekday',
+    label: 'Weekday',
+    description: 'Use the full weekday name, e.g. "Monday".',
+    type: FieldType.STRING,
+    required: false,
+  },
+  monthday: {
+    id: 'monthday',
+    label: 'Monthday',
+    description: 'Number of the day of month, e.g. "27th".',
+    type: FieldType.STRING,
+    required: false,
+  },
+  recurrence_ends: {
+    id: 'recurrence_ends',
+    label: 'Recurrence ends',
+    description: 'Set to true if the recurrence should have an end date.',
+    type: FieldType.BOOLEAN,
+    required: false,
+  },
+  ends_on: {
+    id: 'ends_on',
+    label: 'Ends on',
+    description: 'Recurrence end date in the YYYY-MM-DD format.',
+    type: FieldType.DATE,
```

Comment:
FieldType.DATE means that we either accept a static string (so format will be correct) or a data point with the date in ISO string format? We should be able to handle both of these -- is there a helper we can use to parse the string provided, identify if it is ISO string or 'YYYY-MM-DD' (FHIR date format) and then convert it (if needed) before we use it? If not, I think we will need one for this

Reply:
I'm not sure I understand correctly your questions. But I can tell that it will work with any working date, even more formats than those you listed, because it uses `DateOnlySchema` from `extensions-core`. If validation is successful, the date is converted to `YYYY-MM-DD` format. So basically if you put any valid date here, it will be parsed into `YYYY-MM-DD`.

I'm not sure if that answers your questions.

Reply:
Ah okay, so as long as it's a parseable date then the exact format of the value provided doesn't matter. Understood, that's good to know -- I didn't realise it could handle this automatically, that's good news